sequencer/src/api/fs.rs

```diff
@@ -33,9 +33,9 @@ impl SequencerDataSource for DataSource {
         let mut index = Index::default();
 
         // Index blocks by timestamp.
-        let mut blocks = data_source.get_block_range(..).await?;
+        let mut blocks = data_source.get_block_range(..).await;
         while let Some(block) = blocks.next().await {
-            index_block_by_time(&mut index.blocks_by_time, &block?);
+            index_block_by_time(&mut index.blocks_by_time, &block.await);
```

Comment:
This `await` will block until the block is fetched, if the block was not available immediately. For the FS backend, fetching isn't really supported at all. It's probably better here to keep the old behavior of failing if the block is not available, with `&block.try_resolve()?`